pkg/autoscaler/kubernetes_client.go

```diff
@@ -0,0 +1,236 @@
+	}
+	clusterStatus.totalNodes = int32(len(nodes.Items))
+	var tc resource.Quantity
+	var sc resource.Quantity
+	for _, node := range nodes.Items {
+		tc.Add(node.Status.Capacity[apiv1.ResourceCPU])
+		if !node.Spec.Unschedulable {
+			clusterStatus.schedulableNodes++
+			sc.Add(node.Status.Capacity[apiv1.ResourceCPU])
+		}
+	}
+
+	tcInt64, tcOk := tc.AsInt64()
+	scInt64, scOk := sc.AsInt64()
+	if !tcOk || !scOk {
+		return ClusterStatus{}, fmt.Errorf("unable to compute integer values of schedulable cores in the cluster")
+	}
+	clusterStatus.totalCores = int32(tcInt64)
+	clusterStatus.schedulableCores = int32(scInt64)
+	return clusterStatus, nil
+}
+
+// GetScaler returns a ScalerInterface for client's target resource.
+func (k *KubernetesClient) GetScaler() (ScalerInterface, error) {
+	switch k.target.targetType {
+	case "deployment":
+		return &DeploymentScaler{k}, nil
+	case "rc":
+		return &ReplicationControllerScaler{k}, nil
+	case "rs":
+		return &ReplicaSetScaler{k}, nil
+	}
+	return nil, fmt.Errorf("no scaler has been implemented for %q", k.target.targetType)
+}
+
+// ScalerInterface provides an interface for resources that can be scaled.
+type ScalerInterface interface {
+	// GetNumOfReplicas Get number of replicas of the resource
+	GetNumOfReplicas() (int32, error)
+	// UpdateNumOfReplicas Update the number of replicas for the resource
+	UpdateNumOfReplicas(replicas int32) error
+}
+
+// DeploymentScaler implements ScalerInterface.
+type DeploymentScaler struct {
+	k *KubernetesClient
+}
+
+func (scaler *DeploymentScaler) GetNumOfReplicas() (int32, error) {
+	deployment, err := scaler.k.clientset.ExtensionsClient.Deployments(scaler.k.namespace).Get(scaler.k.target.targetName)
+	if err != nil {
+		return 0, err
+	}
+	return int32(*deployment.Spec.Replicas), nil
+}
+
+func (scaler *DeploymentScaler) UpdateNumOfReplicas(replicas int32) error {
+	if replicas == 0 {
+		return fmt.Errorf("cannot update to 0 replicas")
+	}
+
+	deployment, err := scaler.k.clientset.ExtensionsClient.Deployments(scaler.k.namespace).Get(scaler.k.target.targetName)
+	if err != nil {
+		return err
+	}
+	*deployment.Spec.Replicas = replicas
+	_, err = scaler.k.clientset.ExtensionsClient.Deployments(scaler.k.namespace).Update(deployment)
+
+	return err
+}
+
+// ReplicationControllerScaler implements ScalerInterface.
+type ReplicationControllerScaler struct {
+	k *KubernetesClient
+}
+
+func (scaler *ReplicationControllerScaler) GetNumOfReplicas() (int32, error) {
+	rc, err := scaler.k.clientset.CoreClient.ReplicationControllers(scaler.k.namespace).Get(scaler.k.target.targetName)
+	if err != nil {
+		return 0, err
+	}
+	return int32(*rc.Spec.Replicas), nil
+}
+
+func (scaler *ReplicationControllerScaler) UpdateNumOfReplicas(replicas int32) error {
+	if replicas == 0 {
+		return fmt.Errorf("cannot update to 0 replicas")
+	}
+
+	rc, err := scaler.k.clientset.CoreClient.ReplicationControllers(scaler.k.namespace).Get(scaler.k.target.targetName)
```

Comment:
This is doing it the hard way.  For all of these types, you should be able to call:

```
scale, err := scaler.k.clientset.Extensions().Scales(scaler.k.namespace).Get(kind, name)

scale.spec.Replicas++
_, err = scaler.k.clientset.Extensions().Scales(scaler.k.namespace).Update(kind, scale)
```